lib/fluent/supervisor.rb

```diff
@@ -35,23 +37,109 @@
 end
 
 module Fluent
-  class Supervisor
-    def self.get_etc_passwd(user)
-      if user.to_i.to_s == user
-        Etc.getpwuid(user.to_i)
-      else
-        Etc.getpwnam(user)
-      end
+  module ServerModule
+    def before_run
+      log_opts = {suppress_repeated_stacktrace: config[:suppress_repeated_stacktrace]}
+      log = Supervisor::LoggerInitializer.new(config[:log_path], config[:log_level], config[:chuser], config[:chgroup], log_opts)
```

Comment:
```$log``` is initialized in ```log.init``` and pass this ```$log``` to serverengine.
If it cause confusion, I'll change the name.